modnet/fit_genetic.py

```diff
@@ -70,13 +124,25 @@ def initialization_population(
         """
 
         self.pop =  [[]]*size_pop
-        activation = ['relu', 'elu']
-        loss = ['mae', 'mse']
-        xscale = ['minmax', 'normal']
-        lr = [0.03, 0.01, 0.003, 0.001, 0.0003, 0.0001]
-        initial_batch_size = [8, 16, 32, 64, 128]
-        fraction = [1, 0.75, 0.5, 0.25]
-        self.pop = [[10*randint(1,int(len(self.X_train.get_optimal_descriptors())/10)), 32*randint(1,10), random.choice(fraction), random.choice(fraction), random.choice(fraction), random.choice(activation), random.choice(loss), random.choice(xscale), random.choice(lr), random.choice(initial_batch_size)] for i in range(0, size_pop)]
+        activation = ['elu']
+        loss = ['mae']
+        xscale = ['minmax', 'standard']
+        self.lr = [0.01, 0.005, 0.001]
+        self.initial_batch_size = [8, 16, 32, 64, 128]
+        self.fraction = [1, 0.75, 0.5, 0.25]
+
+        n_features = 0 #initialization
+        if len(self.data.get_optimal_descriptors()) <= 100:
+            b = int(len(self.data.get_optimal_descriptors())/2)
+            n_features = randint(1, b) + b
+        elif len(self.data.get_optimal_descriptors()) > 100 and len(self.data.get_optimal_descriptors()) < 2000:
+            max = len(self.data.get_optimal_descriptors())
+            n_features = 10*randint(1,int(max/10))
+        else:
+            max = np.sqrt(len(self.data.get_optimal_descriptors()))
+            n_features = randint(1,max)**2
+        
+        self.pop = [[n_features , 32*randint(1,10), random.choice(self.fraction), random.choice(self.fraction), random.choice(self.fraction), random.choice(activation), random.choice(loss), random.choice(xscale), random.choice(self.lr), random.choice(self.initial_batch_size)] for i in range(0, size_pop)]
```

Comment:
As @ml-evs suggested I would make the hyperparmeters names more explicit, this will make things more readable (instead of lists indices). For instance by defining a small class 'individual' containing each of the tuned hyperparameters (individual.n_feat etc.).